src/components/molecules/ScheduleItemNG/ScheduleItemNG.tsx

```diff
@@ -1,46 +1,147 @@
-import React from "react";
+import React, { useCallback, useMemo } from "react";
+import classNames from "classnames";
 import { differenceInCalendarDays } from "date-fns";
 
+import { EVENT_LIVE_RANGE } from "settings";
+
 import { PersonalizedVenueEvent } from "types/venues";
+import { Room } from "types/rooms";
 
-import { eventEndTime, eventStartTime } from "utils/event";
+import { eventEndTime, eventStartTime, isEventStartingSoon } from "utils/event";
 import { formatDateRelativeToNow, formatTimeLocalised } from "utils/time";
+import {
+  enterVenue,
+  openUrl,
+  getUrlParamFromString,
+  getUrlWithoutTrailingSlash,
+  getLastUrlParam,
+  getFullVenueInsideUrl,
+} from "utils/url";
+
+import { useShowHide } from "hooks/useShowHide";
+import { useRelatedVenues } from "hooks/useRelatedVenues";
+import { useRoom } from "hooks/useRoom";
+
+import Button from "components/atoms/Button";
 
 import "./ScheduleItemNG.scss";
 
 export interface ScheduleItemNGProps {
   event: PersonalizedVenueEvent;
 }
 
-const formatDateOptions = { formatToday: () => "" };
-
 export const ScheduleItemNG: React.FC<ScheduleItemNGProps> = ({ event }) => {
+  const { currentVenue: eventVenue } = useRelatedVenues({
+    currentVenueId: event.venueId,
+  });
+  const eventRoom = useMemo<Room | undefined>(
+    () =>
+      eventVenue?.rooms?.find((room) => {
+        const { room: eventRoom = "" } = event;
+        const noTrailSlashUrl = getUrlWithoutTrailingSlash(room.url);
+
+        const [roomName] = getLastUrlParam(noTrailSlashUrl);
+        const roomUrlParam = getUrlParamFromString(eventRoom);
+        const selectedRoom = getUrlParamFromString(room.title) === eventRoom;
+
+        return roomUrlParam.endsWith(`${roomName}`) || selectedRoom;
+      }),
+    [eventVenue, event]
+  );
+  const { isShown: isEventExpanded, toggle: toggleEventExpand } = useShowHide();
+  const { enterRoom } = useRoom({
+    room: eventRoom,
+    venueName: eventVenue?.name ?? "",
+  });
   const showDate = Boolean(
     differenceInCalendarDays(eventEndTime(event), eventStartTime(event))
   );
+  const isCurrentEventLive = isEventStartingSoon(event, 2 * EVENT_LIVE_RANGE);
+  const roomUrlParam = getUrlParamFromString(event.room ?? "");
+
+  const handleCopyEventLink = useCallback(
+    (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.stopPropagation();
+
+      const eventLink = getFullVenueInsideUrl(roomUrlParam);
+      navigator.clipboard.writeText(eventLink);
+    },
+    [roomUrlParam]
+  );
+
+  const goToEventLocation = useCallback(() => {
+    if (!eventRoom) {
+      openUrl(roomUrlParam);
+
+      return;
+    }
+
+    if (event.room) {
+      enterRoom();
+    } else {
+      enterVenue(event.venueId);
+    }
+  }, [enterRoom, event, eventRoom, roomUrlParam]);
```

Comment:
Instead of redirecting to the venue, we wanted the room modal to show when you click the button. Can you please make it work like that instead?

Reply:
discussed with @sunny-viktoryia Right now it's better to open the room and leave this behavior as it was in event modal